### PR TITLE
Updates to Contribute section

### PR DESCRIPTION
**Pull request by jasonblais**
This might be preliminary, but I started looking around the contribution section, updating a few of the pages and felt we're pretty close to being ready to ship this.

I've listed below the tasks that would be needed to ship it. Let me know if others are required.

Moreover, treat all of my changes as suggestions. I just brainstormed and prepared the changes, so I’m between 0/5 to 3/5 on most of these. Please be as critical as you'd like :P 

Also closes https://github.com/mattermost/mattermost-developer-documentation/issues/66

Punchlist to ship Contribute section live:

- [X] Separate contributor flow for each project 
  - [X] Condense contributor guidelines in Getting Started section, and turn it to a checklist
     - [X] Add a separate page for contributing without a Help Wanted ticket
  - [X] Create contributor guidelines for mattermost-server
  - [X] Create contributor guidelines for mattermost-webapp
  - [X] Create contributor guidelines for mattermost-redux
  - [X] Create contributor guidelines for mattermost-mobile
  - [X] Create contributor guidelines for desktop
- [X] Update Getting Started section
  - [X] Review intro
  - [X] Update core committers page
  - [X] Update project repositories
- [ ] Finish Desktop App section
  - [x] Update Developer environment setup with recent build changes
  - [ ] Add developer workflow 
- [x] Finish Mobile section
  - [x] Update intro
  - [x] Update Developer environment setup with recent build changes
  - [x] Add developer workflow 
- [X] Finish Server section
  - [x] Update intro
  - [X] Add developer workflow
- [X] Finish Webapp section
  - [x] Update intro
  - [X] Add developer workflow
- [X] Turn all pages public
- [X] Redirect dev docs from docs.mattermost.com to developers.mattermost.com https://github.com/mattermost/docs/pull/2037
  - [X] Contribution guidelines
  - [X] Developer flow
  - [ ] Developer environment setup
  - [ ] Mobile environment setup
  - [X] Core committers
- [ ] Rename platform binary to mattermost in all developer set up docs

### CODE REVIEW

File: site/content/contribute/getting-started/_index.md [hunk 1]
```diff
@@ -9,20 +8,24 @@ section: "contribute"
...
 * [Redux](/contribute/redux/) - The Redux-based web, storage and logic driver of the web and mobile apps
 
-There are some smaller parts of the project not listed here, [see the project repositories page for a full list](/contribute/getting-started/respositories/).
+[See the project repositories page](/contribute/getting-started/respositories/) for a full list of open projects.
 
-Each part of the project will have it's own developer setup instructions, your next step should be to set up your machine using one of those.
+3. **Set up your developer machine**. Each project has its own developer setup instructions.
 
-Unsure where you want to contribute? [Try starting with the server](/contribute/server/developer-setup/).
+4. **Select a [ticket](https://github.com/mattermost/mattermost-server/issues?q=is%3Aissue+is%3Aopen+label%3A%22Up+For+Grabs%22)** and comment to let people know you’re working on it. If there’s no ticket for what you want to work on, follow the process [here](https://docs.mattermost.com/developer/contribution-guide.html#pull-requests-for-minor-corrections-and-improvements-without-a-ticket-are-welcome).
```
Comment (jasonblais) [inline site/content/contribute/getting-started/_index.md]: Fix link to /getting-started/contributions-without-ticket

File: site/content/contribute/getting-started/_index.md [hunk 2]
```diff
@@ -1,5 +1,4 @@
 ---
-draft: true
 title: "Getting Started"
```
Comment (jwilander) [inline site/content/contribute/getting-started/_index.md]: @jasonblais thoughts about making this page a lot simpler, similar to the integrate section?
https://developers.mattermost.com/integrate/getting-started/

File: site/content/contribute/getting-started/contribution-checklist.md [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+date: 2017-08-20T12:33:36-04:00
+weight: 1
+subsection: Getting Started
+---
+
+# Contribution Checklist
+
+Thanks for your interest in contributing to Mattermost! Come [join our "Contributors" community channel](https://pre-release.mattermost.com/core/channels/tickets) on our daily build server, where you can discuss questions with community members and the Mattermost core team.
+
+To help with translations, [see the localization process](https://docs.mattermost.com/developer/localization.html), follow this checklist for submitting a pull request (PR):
```
Comment (jwilander) [inline site/content/contribute/getting-started/contribution-checklist.md]: The first part of the sentence is about translations and the second part is about submitting a PR, seems a bit split-brained. Maybe divide them into two sentences and put them on different lines?

File: site/content/contribute/getting-started/contributions-without-ticket.md [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+It is acceptable to submit a PR for a bug or incremental improvement, with less than 20 lines of code change, without a Help Wanted ticket if the change to existing behaviour is minor. All pull requests submitted without a corresponding ticket will first be reviewed by a core team product manager. Some examples of minor corrections and improvements include:
+
+ - [Fix a formatting error in help text](https://github.com/mattermost/mattermost-server/pull/5640)
+ - [Fix success typo in Makefile](https://github.com/mattermost/mattermost-server/pull/5809)
+ - [Fix broken Cancel button in Edit Webhooks screen](https://github.com/mattermost/mattermost-server/pull/5612)
+ - [Fix Android app crashing when saving user notification settings](https://github.com/mattermost/mattermost-mobile/pull/364)
+ - [Fix recent mentions search not working](https://github.com/mattermost/mattermost-server/pull/5878)
+ 
+Larger pull requests, including those that require a change to documentation or user expectations, [require a Help Wanted issue opened by the core team](http://docs.mattermost.com/process/help-wanted.html) so that the change can be tested, documented and supported.
```
Comment (jwilander) [inline site/content/contribute/getting-started/contributions-without-ticket.md]: >tested, documented and supported

That sounds like the BS reason. We can do all that after a PR is submitted. The real reason to me is we want the product to have a unified vision, and to do that large changes need to discussed and agreed upon by the Mattermost core team.

The text that was there before is much more genuine IMO:

>Core committers who review the PR are entitled to reject it if they feel it significantly changes behavior or user expectations, which requires a Help Wanted ticket opened by the core team. This helps ensure that everything going into the project aligns with a unified vision.

File: site/content/contribute/getting-started/respositories.md [hunk 5]
```diff
@@ -1,5 +1,4 @@
 ---
-draft: true
 title: "Project Respositories"
```
Comment (jwilander) [inline site/content/contribute/getting-started/respositories.md]: I'm unsure if we should keep this page or not. Don't know if it's helpful

**Comment by jwilander** [comment]
Awesome work! This gets us real close to being to go live with this section

I added some comments and push some minor changes directly

File: site/content/contribute/getting-started/_index.md (hunk 2 above)
Comment (jasonblais) [reply site/content/contribute/getting-started/_index.md]: Great, I forgot we were already using that template. It'll be easy to migrate over to that.

Only question is: For step 2, I'm aiming to bring people right away to the project-specific contributor flow, with environment set up, etc. Should we keep that here, or how would we  go about it? 0/5

File: site/content/contribute/getting-started/contribution-checklist.md (hunk 3 above)
Comment (jasonblais) [reply site/content/contribute/getting-started/contribution-checklist.md]: Yeah, I'll fix that. It's confusing as-is.

File: site/content/contribute/getting-started/contributions-without-ticket.md (hunk 4 above)
Comment (jasonblais) [reply site/content/contribute/getting-started/contributions-without-ticket.md]: I just copy-pasted what we had. We can update that. Wasn't aware (or had forgotten) about the updated text.

**Comment by jasonblais** [comment]
> This gets us real close to being to go live with this section

Is the rest captured in the punchlist in PR description, or is there something else I should add?

File: site/content/contribute/getting-started/_index.md (hunk 2 above)
Comment (jwilander) [reply site/content/contribute/getting-started/_index.md]: Yeah I think keeping it here makes sense

File: site/content/contribute/getting-started/contributions-without-ticket.md (hunk 4 above)
Comment (jwilander) [reply site/content/contribute/getting-started/contributions-without-ticket.md]: Ah ok, I thought it was new text you had written 👍

**Comment by jwilander** [comment]
Nope, the PR description covers it

**Comment by jasonblais** [comment]
@jwilander Wondering if you were interested taking over from here? Addressing your feedback in the PR (I agree with all of it) + queue a couple of tickets to finish the sections.

We can probably rally other people outside of ABC involved with the desktop/mobile sections. I'm not actually sure if there's much left for them, look pretty complete to me.

Thoughts?

I can finish the docs.mattermost.com stuff once the dev flows are completed.

**Comment by jwilander** [comment]
@jasonblais I think is pretty much ready for review. I've made a bunch of updates and migrated all relevant pages from docs.mattermost.com. Only thing we're really missing is the desktop app workflow

**Comment by jasonblais** [comment]
Awesome, let's discuss during our 1-1

File: site/content/contribute/desktop/_index.md [hunk 6]
```diff
@@ -5,35 +5,16 @@ section: "contribute"
 ---
 
 # Desktop App
-The Mattermost desktop app is an [Electron](https://electronjs.org/) wrapper around the <a href="/contribute/webapp">webapp</a> project. It lives in the [mattermost/desktop](https://github.com/mattermost/desktop) repository. The desktop app runs on Windows, Linux, and macOS.
+The Mattermost desktop app is an [Electron](https://electronjs.org/) wrapper around the <a href="/contribute/webapp">webapp</a> project. It lives in the [mattermost/desktop]() repository. The desktop app runs on Windows, Linux, and macOS.
```
Comment (yuya-oc) [inline site/content/contribute/desktop/_index.md]: URL is missing.

`[mattermost/desktop]()` -> `[mattermost/desktop](https://github.com/mattermost/desktop)`

**Comment by yuya-oc** [comment]
https://github.com/mattermost/mattermost-developer-documentation/blob/a38bdcf03a95167a014468ab597958c0df009755/site/content/contribute/desktop/developer-setup.md#prerequisites

On "Prerequisites" section, Node.js should be upgraded: `Node.js 8 or later`.

File: site/content/contribute/desktop/_index.md (hunk 6 above)
Comment (jwilander) [reply site/content/contribute/desktop/_index.md]: Nice catch, will update

**Comment by jwilander** [comment]
Thanks @yuya-oc! Updated

**Pull request merged**